lib/scanner/xml-scanner/xml-scanner.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#ifndef XML_SCANNER_H_INCLUDED
+#define XML_SCANNER_H_INCLUDED
+
+#include "syslog-ng.h"
+#include "messages.h"
+
+#include <string.h>
+
+typedef struct _XMLScanner XMLScanner;
+
+typedef void (*PushCurrentKeyValueCB)(const gchar *name, const gchar *value, gssize value_length, gpointer user_data);
+
+typedef gboolean (*StartElement) (XMLScanner *self, const gchar *element_name, const gchar **attribute_names,
```

Comment:
I feel these names too general for a public api. I would call XMLScannerStartElement.

Reply:
They are used through `xml_scanner_end_element_method` API, and they are protected rather than public.

Reply:
If they are just virtual methods for XMLScanner descendants, you can just embed them into the scanner without typedefing them.

Reply:
I've added the typedefs to save space and avoid multi-line definitions inside the struct definition.

You found the typedefs confusing, as it indicates it is a public API, right?
If this would be in the .c file I think the question would not arise.

I'll do @MrAnno's request as it seems to be a consensus.

Reply:
Done in 04d36c9